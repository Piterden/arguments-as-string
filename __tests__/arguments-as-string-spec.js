'use strict'

/* eslint-env jest */
const argumentsAsString = require('..')

describe('arguments-as-string', () => {
  it('is a function', () => {
    expect(typeof argumentsAsString).toBe('function')
  })

  it('forms a string from strings and numbers', () => {
    expect(argumentsAsString('foo', 'bar', 42, 'the last argument')).toMatchSnapshot()
  })

  it('forms string from objects', () => {
    const o = {
      foo: {
        bar: 42
      }
    }
    const p = {
      name: 'Joe'
    }
    const result = argumentsAsString(o, p)
    expect(result).toMatchSnapshot()
  })

  it('handles circular references', () => {
    const o = {
      foo: {
        bar: 42
      }
    }
    const p = {
      name: 'Joe',
    }
    p.ref = p
    const result = argumentsAsString(o, p)
    expect(result).toMatchSnapshot()
  })

  it('handles numbers', () => {
    const result = argumentsAsString(1, 2, 3, 4, 5, 6)
    expect(result).toMatchSnapshot()
  })
})
