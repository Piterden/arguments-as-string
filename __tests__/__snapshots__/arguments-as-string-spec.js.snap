// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arguments-as-string forms a string from strings and numbers 1`] = `"foo bar 42 the last argument"`;

exports[`arguments-as-string forms string from objects 1`] = `
"{
  \\"foo\\": {
    \\"bar\\": 42
  }
} {
  \\"name\\": \\"Joe\\"
}"
`;

exports[`arguments-as-string handles circular references 1`] = `
"{
  \\"foo\\": {
    \\"bar\\": 42
  }
} { cannot stringify arg 1, it has type \\"object\\" with keys name, ref }"
`;

exports[`arguments-as-string handles numbers 1`] = `"1 2 3 4 5 6"`;
